plugins/inventory/vmware_vm_inventory.py

```diff
@@ -314,41 +335,112 @@ def check_requirements(self):
             raise AnsibleError("Missing one of the following : hostname, username, password. Please read "
                                "the documentation for more information.")
 
-    @staticmethod
-    def _process_object_types(vobj, level=0):
-        """For an object that is not a valid JSON type, loop over its properties
-        and return them in a dictionary"""
-        try:
-            json.dumps(vobj)
-            return vobj
-        except Exception:
-            rdata = {}
-            properties = dir(vobj)
-            properties = [str(x) for x in properties if not x.startswith('_')]
-            properties = [x for x in properties if x not in ('Array', 'disabledMethod', 'declaredAlarmState')]
-            properties = sorted(properties)
-
-            for prop in properties:
-                # Attempt to get the property, skip on fail
-                try:
-                    propToSerialize = getattr(vobj, prop)
-                except Exception:
-                    continue
-
-                if callable(propToSerialize):
-                    continue
-
-                prop = prop.lower()
-                if level + 1 <= 2:
-                    try:
-                        rdata[prop] = BaseVMwareInventory._process_object_types(
-                            propToSerialize,
-                            level=(level + 1)
-                        )
-                    except vim.fault.NoPermission:
-                        pass
-
-        return rdata
+    def _get_managed_objects_properties(self, vim_type, properties=None, resources=None, strict=False):
```

Comment:
We got rid of this method since we are using `vcenter.VM.list(vm_filter_spec)` method from `vSphere Automation SDK`. I would like to stick to that since it comparatively faster than `Pyvmomi` methods.

Reply:
In terms of performance `PyVmomi` with `PropertySpec` is way faster. Because it can query wanted properties in very few server requests. Unlike vSphere Automation SDK, `VM.List` return VMs list only and then a number of request requested per properties per VM. I did performance test and see the number of request.


The following test done on vCenter with 5 VMs. 

### With default(15) properties:

```bash

## vSphere Automation SDK (1,897 requests/21.292s)
$ time python ansible-inventory-patched.py -i ../inventory/hci.vmware.yaml --list  2>/dev/null | grep "^send" | wc -l 
    1897

real    0m21.292s
user    0m9.902s
sys     0m0.375s

## PyVmomi (45 requests/1.291s)
$ time python ansible-inventory-patched.py -i ../inventory/hci.vmware.yaml --list  2>/dev/null | grep "^send" | wc -l 
      45

real    0m1.291s
user    0m0.807s
sys     0m0.130s
```


### With all properties:

```bash
## vSphere Automation SDK (18,837 requests/3m49.578s)
$ time python ansible-inventory-patched.py -i ../inventory/hci.vmware.yaml --list  2>/dev/null | grep "^send" | wc -l 
   18837

real    3m49.578s
user    1m42.944s
sys     0m1.736s


## PyVmomi (65 requests/0m1.678s)
$ time python ansible-inventory-patched.py -i ../inventory/hci.vmware.yaml --list  2>/dev/null | grep "^send" | wc -l 
      65

real    0m1.678s
user    0m0.942s
sys     0m0.166s
```

---
```py
# ansible-inventory-patched.py
import logging
from http.client import HTTPConnection
import sys
from ansible import context
from ansible.errors import AnsibleError, AnsibleOptionsError, AnsibleParserError
from ansible.module_utils._text import to_text


log = logging.getLogger('urllib3')
log.setLevel(logging.DEBUG)
ch = logging.StreamHandler()
ch.setLevel(logging.DEBUG)
log.addHandler(ch)
HTTPConnection.debuglevel = 1

sub = 'inventory'
myclass = "%sCLI" % sub.capitalize()

mycli = getattr(__import__("ansible.cli.%s" % sub, fromlist=[myclass]), myclass)
args = [to_text(a, errors='surrogate_or_strict') for a in sys.argv]
cli = mycli(args)
exit(cli.run())
```

Reply:
@Akasurde your thoughts?

Reply:
I reverted my enhancements (https://github.com/ansible-collections/vmware/pull/85) since this PR looks promising.